### PR TITLE
add windows_prefered_cluster option to cluster schema/structure

### PR DESCRIPTION
**Pull request by zparnold**
This PR adds the ability to set the `windows_prefered_cluster` field 

It also copies the spelling from the `cluster_template` files. (The word "preferred" is misspelled.) I wanted to keep it consistent however. If I should fix the spelling, please let me know.

### CODE REVIEW

File: rancher2/structure_cluster.go [hunk 1]
```diff
@@ -179,6 +179,11 @@ func flattenCluster(d *schema.ResourceData, in *Cluster, clusterRegToken *manage
 		return err
 	}
 
+	err = d.Set("windows_prefered_cluster", in.WindowsPreferedCluster)
+	if err != nil {
+		return err
+	}
+
```
Comment (rawmind0) [inline rancher2/structure_cluster.go:183]: Bool fields set don't need to be checked

```
-	err = d.Set("windows_prefered_cluster", in.WindowsPreferedCluster)
-	if err != nil {
-		return err
-	}
+	d.Set("windows_prefered_cluster", in.WindowsPreferedCluster)
```

File: rancher2/schema_cluster.go [hunk 2]
```diff
@@ -274,6 +274,12 @@ func clusterFields() map[string]*schema.Schema {
 			Optional: true,
 			Computed: true,
 		},
+		"windows_prefered_cluster": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "Windows prefered cluster",
```
Comment (rawmind0) [inline rancher2/schema_cluster.go]: As rancher/types field name is `WindowsPreferedCluster` to be consistent don't need to change tf field name, but description should be updated

```
-			Description: "Windows prefered cluster",
+			Description: "Windows preferred cluster",
```

**Review by rawmind0 — CHANGES_REQUESTED** [review]
@zparnold thanks for the PR. 

Could you please take a look to changes requested?? 

Also, could you please update `rancher2_cluster` resource docs properly, adding new field??

File: rancher2/structure_cluster.go (hunk 1 above)
Comment (zparnold) [reply rancher2/structure_cluster.go:183]: Sounds good 😄

File: rancher2/schema_cluster.go (hunk 2 above)
Comment (zparnold) [reply rancher2/schema_cluster.go]: 👍

**Comment by zparnold** [comment]
@rawmind0 Made changes you requested. Also did a force push so there will only be one commit when merged. 😄 Let me know if there's anything else I need to do.

**Review by rawmind0 — CHANGES_REQUESTED** [review]
@zparnold thanks for the update. 

Last thing, could you please add new field to be updated on `resourceRancher2ClusterUpdate` function on `resource_rancher2_cluster.go` and to be tested on `structure_cluster_test.go`??

**Comment by zparnold** [comment]
@rawmind0 Addressed your concerns, and welcome your feedback.

File: rancher2/resource_rancher2_cluster.go [hunk 3]
```diff
@@ -398,3 +402,12 @@ func createClusterRegistrationToken(client *managementClient.Client, clusterID s
...
 }
+
+func windowsPreferredClusterCustomDiffHandler()  schema.CustomizeDiffFunc{
+	//Cluster must be recreated to implement this if there is a change to windows var if the cluster is not
+	//already a windows cluster. Currently this is unsupported by rancher, so force new resource here.
+	//https://rancher.com/docs/rancher/v2.x/en/cluster-provisioning/rke-clusters/windows-clusters/
+	return customdiff.ForceNewIfChange("windows_prefered_cluster", func (old, new, meta interface{}) bool {
+		return new.(bool) != old.(bool)
+	})
+}
```
Comment (rawmind0) [inline rancher2/resource_rancher2_cluster.go]: If a new object needs to be generated on this field change, i think is just enough to add `Forcenew: true` on the field definition. Not needed a cutomizediff function.

**Review by rawmind0 — CHANGES_REQUESTED** [review]
Please, take a look on the new comments.

Also, build is failing, please execute `make fmt` and  `make test` before commit.

File: rancher2/resource_rancher2_cluster.go (hunk 3 above)
Comment (zparnold) [reply rancher2/resource_rancher2_cluster.go]: Gotcha, thanks!

**Comment by zparnold** [comment]
@rawmind0 Hopefully this should address that concern.

**Review by rawmind0 — APPROVED** [review]

**Comment by rawmind0** [comment]
LGTM

**Pull request merged**